_data-prepper/pipelines/configuration/processors/key-value.md

```diff
@@ -10,20 +10,20 @@ nav_order: 45
+| field_delimiter_regex | A regular expression specifying the delimiter that separates key-value pairs. Special characters such as `[` and `]` must be escaped with `\\`. This option cannot be defined at the same time as `field_split_characters`. Required conditionally. No default value. | If `field_delimiter_regex` is `"&\\{2\\}"`, `{"key1=value1&&key2=value2"}` parses into `{"key1": "value1", "key2": "value2"}`. |
+| field_split_characters | A string of characters specifying the delimeter that separates key-value pairs. Special regular expression characters such as `[` and `]` must be escaped with `\\`. This option cannot be defined at the same time as `field_delimiter_regex`. Required conditionally. Default value is `&`. | If `field_split_characters` is `"&&"`, `{"key1=value1&&key2=value2"}` parses into `{"key1": "value1", "key2": "value2"}`. |
+| key_value_delimiter_regex | A regular expression specifying that separates a key and a value within a key-value pair. Special regular expression characters such as `[` and `]` must be escaped with `\\`. This option cannot be defined at the same time as `value_split_characters`. Required conditionally. No default value. | If `key_value_delimiter_regex` is `"=\\{2\\}"`, `{"key1==value1"}` parses into `{"key1": "value1"}`. |
+| value_split_characters | A string of characters specifying the delimiter that separates a key and value within a key-value pair. Special regular expression characters such as `[` and `]` must be escaped with `\\`. This option cannot be defined at the same time as `key_value_delimiter_regex`. Required conditionally. Default value is `=`. | If `value_split_characters` is `"=="`, `{"key1==value1"}` parses into `{"key1": "value1"}`. |
+| non_match_value | When a key-value cannot be successfully split, the key-value pair is placed in the `key` field, and the specified value is placed in the value field. Optional. Default value is `null`. | `key1value1&key2=value2` parses into `{"key1value1": null, "key2": "value2"}`. |
+| prefix | A prefix to append before all keys. Optional. Default value is an empty `string`. | If `prefix` is `"custom"` and `{"key1=value1"}`, it parses into `{"customkey1": "value1"}`.|
+| delete_key_regex | A regular expression specifying the characters to delete from the key. Special regular expression characters such as `[` and `]` must be escaped with `\\`. Optional. No default value. | If `delete_key_regex` is `"\s"`, `{"key1 =value1"}` parses into `{"key1": "value1"}`. |
+| delete_value_regex | A regular expression specifying the characters to delete from the value. Special regular expression characters such as `[` and `]` must be escaped with `\\`. Optional. No default value. | If `delete_value_regex` is `"\s"`, `{"key1=value1 "}` parses into `{"key1": "value1"}`. |
```

Comment:
```suggestion
| delete_value_regex | A regular expression specifying the characters to delete from the value. Special regular expression characters such as `[` and `]` must be escaped with `\\`. Cannot be an empty string. Optional. No default value. | If `delete_value_regex` is `"\s"`, `{"key1=value1 "}` parses into `{"key1": "value1"}`. |
```

Reply:
In my opinion, since there's a lot of text here and the examples are long, it is better to list them in a list as opposed to a table. That way, each point in the description can become a sublist. I much prefer the layout in the repo https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/key-value-processor/README.md. @natebower What do you think?

Reply:
Visually, it is much easier to parse through these options in a tabular format. I am opting to keep the table format for now.